### ISSUE TITLE
Supermarket tries to hit id.chef.io even when authenticating to private chef server

### ISSUE THREAD

**Issue by jf647** [chef/supermarket#989]
I'm seeing failures in my sidekiq log like this:

```
2015-02-27T22:41:47.739Z 9164 TID-ows47yi2g WARN: {"retry"=>true, "queue"=>"default", "class"=>"OauthTokenRefreshWorker", "args"=>[1], "jid"=>"fbd20b0c4704bab37bc5c1f4", "enqueued_at"=>1425074700.0613215, "error_message"=>"Connection timed out - connect(2) for \"id.chef.io\" port 443", "error_class"=>"Errno::ETIMEDOUT", "failed_at"=>1425074827.355629, "retry_count"=>6, "retried_at"=>1425076907.7389627}
```

This supermarket instance is on a non-internet connected AWS VPC, and is set up to authenticate against our Enterprise Chef server (on prem over a VPN).  It should not be trying to hit public sites for any reason.  OAuth to that chef server is working (I can log in, upload cookbooks, etc.)

supermarket calls out to the ChefOAuth2 omniauth provider here:

https://github.com/chef/supermarket/blob/09529bb0d708a356b4ab920428a574bfaf24552c/app/workers/oauth_token_refresh_worker.rb#L12-16

And that gem appears to have defaults that hit id.chef.io:

https://github.com/chef/omniauth-chef-oauth2/blob/master/lib/omniauth/strategies/chef_oauth2.rb#L9-13

But it does not appear that supermarket overrides this with anything from the ENVDIR file.  There doesn't even appear to be any direct reference to id.chef.io in the `.env.production` file.  There is CHEF_DOMAIN:

    CHEF_DOMAIN=chef.io

but if that's how id.chef.io is being derived, then I'm not sure how I'm meant to use it, because that would require me to add a DNS name of id.mychefserver.mycompany.com.

Can someone shed some light on this?  Trying to get access tokens for private users from a public service is not confidence inspiring.

**Comment by smith** [comment]
@jf647 It looks like it should be setting the `:site` key in that options hash in the worker to `ENV['CHEF_OAUTH2_URL']`.

This is done in the initializer (https://github.com/chef/supermarket/blob/62a762af55695a3b9f0a27fab032a5c91683089f/config/initializers/omniauth.rb#L32-L34) but I think it might need to be done here as well.

Not sure how this is supposed to work, but if there's a way to use the same initialized object with the correct settings from the initializer in the worker we should do that. Otherwise we should make it use all of the correct environment variables.

Pretty sure this has been broken all along and you're the first to notice it, so thanks for opening this!

**Comment by leftathome** [comment]
Hmm, I wonder if this means that Sidekiq's been broken for three months for me in production... :O 

Maybe, sort of.  I see this error from that worker:
````
2015-03-03_23:25:23.48231 2015-03-03T23:25:23.482Z 17850 TID-oqzsiciq8 OauthTokenRefreshScheduleWorker JID-e10c85864f88d84c723a3514 INFO: start
2015-03-03_23:25:23.48594 D, [2015-03-03T23:25:23.485876 #17850] DEBUG -- :   (1.3ms)  SELECT "accounts"."id" FROM "accounts" INNER JOIN "users" ON "users"."id" = "accounts"."user_id" WHERE ("accounts"."oauth_expires" BETWEEN '2015-03-03 23:25:00.000000' AND '2015-03-03 23:50:01.000000')
2015-03-03_23:25:23.48749 2015-03-03T23:25:23.487Z 17850 TID-oqzsiciq8 OauthTokenRefreshScheduleWorker JID-e10c85864f88d84c723a3514 INFO: done: 0.005 sec
2015-03-03_23:30:08.36777 2015-03-03T23:30:08.367Z 17850 TID-oqzsiciq8 OauthTokenRefreshWorker JID-a80a5542aed603a1556770f9 INFO: start
2015-03-03_23:30:08.36978 D, [2015-03-03T23:30:08.369713 #17850] DEBUG -- :   ^[[1m^[[36mAccount Load (0.5ms) SELECT  "accounts".* FROM "accounts"  WHERE "accounts"."id" = $1 LIMIT 1  [["id", 35]]
2015-03-03_23:30:08.62718 2015-03-03T23:30:08.627Z 17850 TID-oqzsiciq8 OauthTokenRefreshWorker JID-a80a5542aed603a1556770f9 ERROR: invalid_request: The request is missing a required parameter, includes an unsupported parameter value, or is otherwise malformed.
2015-03-03_23:30:08.62721 {"error":"invalid_request","error_description":"The request is missing a required parameter, includes an unsupported parameter value, or is otherwise malformed."}
2015-03-03_23:30:08.62749 2015-03-03T23:30:08.627Z 17850 TID-oqzsiciq8 OauthTokenRefreshWorker JID-a80a5542aed603a1556770f9 INFO: done: 0.26 sec
````

**Comment by nellshamrell** [comment]
Working on reproducing this right now, after that I'll code up a solution.

**Comment by nellshamrell** [comment]
I am almost ready to release an updated package with this in it, just running some final tests.  It should be available tomorrow, I'll update here when it is.

**Comment by nellshamrell** [comment]
Having some difficulty releasing the updated package, pushing this until Monday.

**Comment by nellshamrell** [comment]
I just released this as version 1.9 of the Supermarket Omnibus package (and also merged it into the master branch in this repository)

Going to close this for now, but please let us know if you see any further issues!
